src/client/WindshaftMetadata.js

```diff
@@ -0,0 +1,178 @@
+//             min: 1,      // actual stats of property _cdb_dim_month_time1
+//             max: 24
+//         }
+//     },
+
+//     // But wait, there's more, we can transfer a date dimension as an iso string property
+//     // but then expose it as two date properties, one for the start of each period and
+//     // one from the end (if not used, any of them may not be present)
+//     // these dimension properties are triggered by clusterMonthStart/clusterMonthEnd etc.
+//     time2: {
+//         type: 'date' // this is the type of the base column only,
+//         min: '2017-01-01T00:00:00',  // stats of the base column
+//         max: '2018-12-01T00:00:00',
+//         dimension: {
+//             type: 'category', // type of the actual (source) property received internally
+//             grouping: {},// (to be changed to group) grouping parameters
+//             propertyName: "_cdb_dim_month_time1", // source property
+//             min: '2017-01', // stats of property _cdb_dim_month_time1
+//             max: '2017-12',
+//             range: [
+//                 // here come the actual decoded (internal) properties!!:
+//                 // the types of these properties is 'date'
+//                 '_cdb_dim_month_time1_start',
+//                 '_cdb_dim_month_time1_end'
+//             }
+//         }
+//     }
+// }
+
+export default class WindshaftMetadata extends MVTMetadata {
+    constructor (...args) {
+        super(...args);
+        this.propertyKeys = [];
+        this.baseNames = {};
+        Object.keys(this.properties).forEach(baseName => {
+            const property = this.properties[baseName];
+            if (property.aggregations) {
+                Object.values(property.aggregations).forEach(propName => {
+                    this._addProperty(baseName, propName);
+                });
+            } else if (property.dimension) {
+                if (property.dimension.range) {
+                    property.dimension.range.forEach(rangePropertyName => {
+                        this._addProperty(baseName, rangePropertyName, false);
+                    });
+                    // add source property too, for stats
+                    this._addProperty(baseName, property.dimension.propertyName);
+                } else {
+                    this._addProperty(baseName, property.dimension.propertyName);
+                }
+            } else {
+                this._addProperty(baseName, baseName);
+            }
+        });
+    }
+
+    _addProperty (baseName, propertyName, addToKeys = true) {
+        this.baseNames[propertyName] = baseName;
+        if (addToKeys) {
+            this.propertyKeys.push(propertyName);
+        }
+    }
+
+    _dimensionInfo (propertyName) {
+        const baseName = this.baseName(propertyName) || propertyName;
+        const column = this.properties[baseName];
+        let dimension = null;
+        if (baseName !== propertyName) {
+            if (column.dimension) {
```

Comment:
No else in this nested if, so why not using...

```suggestion
            if (baseName !== propertyName && column.dimension) {
                // whatever
            }
```